SeaBattle/Sea/Sea.cs

```diff
@@ -57,24 +58,20 @@ public void AddShip(BaseShip s1)
                 Ships.Add(s1);
   
         }
+
         public bool ShipSpaceCheck(List<Point> DecksPoints)
         {
-            foreach (var deck in DecksPoints)
+            foreach (var (deck, point) in from deck in DecksPoints
+                                          from ship in Ships
+                                          from point in ship.Decks
+                                          select (deck, point))
             {
-                foreach (var ship in Ships)
-                {
-                    foreach (var point in ship.Decks)
-                    {
-                        // If this true - point is occupied so thats why it returns false 
-                        if (deck.X == point.Location.X || deck.Y == point.Location.Y)
-
-                            return false;
-                        else 
-                            return true;
-                    }
-                }
+                // If this true - point is occupied so thats why it returns false 
+                if (deck.X == point.Location.X || deck.Y == point.Location.Y)
```

Comment:
You can simplify this if/else by returning the statement - return deck.X == point.Location.X || deck.Y == point.Location.Y;

Reply:
Again, because of naming, it's hard to understand what this method should return, is it true if the place is occupied? Or maybe it's false, cause you can't place a new one here? You better name it something like "IsPlaceOccupied" then true will answer on this question - yes, it's occupied. Besides that, your check with | | is wrong, there has to be a && check, moreover the Point class has == operator overridden so you can just compare them like this deck == point. Regarding your changes, it won't work like that, did you test it at all? Btw you don't have a check for decks coords which are not aligned, for example in your system ship with decks like Decks{ {1,1}, {1,2}, {3,5} } is valid.